### ISSUE TITLE
Unable to type form values

### ISSUE THREAD

**Issue by mattoni** [grommet/grommet#4840]
### Expected Behavior

I should be able to pass a generic type to the form, i.e. `<Form<FormValueType>>`, in order to get proper typing for `onSubmit` etc

### Actual Behavior

It's not possible to type the form, and passing in external onSubmit handlers have type errors if you attempt to type the params manually.

### URL, screen shot, or Codepen exhibiting the issue

<img width="492" alt="image" src="https://user-images.githubusercontent.com/5110855/102720697-aed4fe80-42aa-11eb-9d19-592b099e9591.png">

<img width="989" alt="image" src="https://user-images.githubusercontent.com/5110855/102720703-bac0c080-42aa-11eb-8950-2139909dfb12.png">

### Steps to Reproduce

1. Attempt to pass a typed onSubmit handler to the <Form> component

### Your Environment

- Grommet version: 2.16.2

**Comment by kenjdavidson** [comment]
You need to create a typed form. Don't mind my comments, but it took a while.

```
// Super intuitive and not documented anywhere but a commit message
// https://github.com/grommet/grommet/commit/ed480a02d0fb053e9466bc137541e436e1e77c15
const TypedCourseForm = Form as TypedForm<Course>;
```

But it works great.

**Comment by mattoni** [comment]
That does indeed work, though it seems a bit redundant to do instead of what most other libraries do, i.e. `Form<FormProps>`. I've opened a PR to allow that method, in case there is more interest.  #4841

**Comment by ShimiSun** [comment]
@mattoni Thank you for your contribution, the fix will be available on our next release (later than v2.16.3)
